src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs

```diff
@@ -198,14 +200,22 @@ public void DialogSendKeys(string dialogAutomationName, string keys)
             SendKeys(new KeyPress(VirtualKey.K, ShiftState.Ctrl), new KeyPress(VirtualKey.F, ShiftState.Ctrl));
         }
 
-        public void Paste(string text)
+        private void ExecuteOnSTAThread(ThreadStart a)
         {
-            var thread = new Thread(() => Clipboard.SetText(text));
+            var thread = new Thread(a);
             thread.SetApartmentState(ApartmentState.STA);
```

Comment:
I don't think making a new random STA thread helps.  Shouldn't we just schedule it directly on the actual VS UI thread?